src/app/_components/PillButton/PillButton.js

```diff
@@ -0,0 +1,70 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { cn } from '@/app/utils/cn';
+import { Paragraph } from '../Typography';
+
+export function PillButton({
+  children,
+  className,
+  type,
+  colorVariant,
+  icon,
+  disabled = false,
+  onClick,
+}) {
+  const buttonType = icon ? type?.icon : type?.regular;
+  const {
+    regular,
+    hover,
+    focused,
+    active,
+    disabled: disabledState,
+  } = colorVariant || {};
+
+  const { buttonStyle, layoutStyle } = buttonType || {};
+
+  const styles = cn(
+    'gap-[8px] rounded-[100px] font-bold',
+    buttonStyle,
+    regular,
+    hover,
+    focused,
+    active,
+    disabledState,
+    className,
+  );
+
+  return (
+    <button className={styles} disabled={disabled} onClick={onClick}>
+      <div className={layoutStyle || ''}>
+        {icon}
+        <Paragraph className={'text-inherit'}>{children}</Paragraph>
```

Comment:
🟢 i think there are extra {} around 'text-inherit'

Reply:
Thanks for comment! Done.